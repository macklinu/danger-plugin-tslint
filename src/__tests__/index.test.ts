import * as path from 'path'

import tslint from '../'

// tslint:disable:no-string-literal

beforeEach(() => {
  global['fail'] = jest.fn()
  global['message'] = jest.fn()
  global['warn'] = jest.fn()
})

describe('default results handler', () => {
  test('handles successful results', () => {
    tslint({
      lintResultsJsonPath: path.resolve(__dirname, '__fixtures__', 'success-results.json'),
    })
    expect(global['message']).toHaveBeenCalled()
  })
  test('handles lint failures', () => {
    tslint({
      lintResultsJsonPath: path.resolve(__dirname, '__fixtures__', 'failure-results.json'),
    })
    expect(global['fail']).toHaveBeenCalled()
  })
  test('warns when cannot find results', () => {
    tslint({
      lintResultsJsonPath: path.resolve(__dirname, '__fixtures__', 'missing-file.json'),
    })
    expect(global['warn']).toHaveBeenCalled()
  })
})

test('throws when config is not supplied', () => {
  expect(() => tslint()).toThrowErrorMatchingSnapshot()
})

test('throws when lintResultsJsonPath is not supplied', () => {
  expect(() => tslint({})).toThrowErrorMatchingSnapshot()
})

test('throws when lintResultsJson is null', () => {
  expect(() => tslint({
    lintResultsJson: null,
  })).toThrowErrorMatchingSnapshot()
})

test('throws when both lintResultsJsonPath and lintResultsJson are supplied', () => {
  const lintResultsJson = [{
    endPosition: {
      character: 1,
      line: 1,
      position: 1,
    },
    failure: 'foo',
    name: 'bar',
    ruleName: 'foo',
    ruleSeverity: 'foo',
    startPosition: {
      character: 1,
      line: 1,
      position: 1,
    },
  }]

  expect(() => tslint({
    lintResultsJson,
    lintResultsJsonPath: '/foo/bar/baz',
  })).toThrowErrorMatchingSnapshot()
})
