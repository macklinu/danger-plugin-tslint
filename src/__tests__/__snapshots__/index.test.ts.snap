// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws when config is not supplied 1`] = `"Configuration not supplied"`;

exports[`throws when lintResultsJsonPath is not supplied 1`] = `"'lintResultsJsonPath' not supplied"`;
